[flang] Add Flang Call Notes documents (#180287)
llvm/llvm-project: flang/docs/MeetingNotes/README.md

<!--===- docs/MeetingNotes/README.md

   Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
   See https://llvm.org/LICENSE.txt for license information.
   SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

-->
# Combined Flang Community Call Notes

## About these notes

This directory combines the notes from the Flang Technical and Community Calls. This was done to make the call notes more easily available than before. All references to Flang in these documents refer to the Fortran compiler being developed as part of the LLVM project and located in the LLVM monorepo ([github.com/llvm/llvm-project](http://github.com/llvm/llvm-project)). Information about "Classic Flang", a separate and independent Fortran compiler, can be found at [github.com/flang-compiler/flang](http://github.com/flang-compiler/flang).

## Call Information

> - Details about the call can be found in [GettingInvolved.md](https://github.com/llvm/llvm-project/blob/main/flang/docs/GettingInvolved.md) under the heading "Flang Biweekly Call"

Reviewer: We could provide a direct link to the appropriate section of the document.

```suggestion
- Details about the call can be found in [GettingInvolved.md](https://flang.llvm.org/docs/GettingInvolved.html#flang-biweekly-call).
```

<!--===- docs/MeetingNotes/README.md

   Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
   See https://llvm.org/LICENSE.txt for license information.
   SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

-->
# Combined Flang Community Call Notes

## About these notes

> This document now combines the notes from the LLVM Flang Technical Calls

Author: I believe that is a much larger debate and outside the scope of this PR.  My intention with this PR was merely to have a better flow for the calls and recording of the meeting minutes.  Perhaps you would like to bring this up on Wednesday's call, or start your own RFC on the subject?

and the LLVM Flang Community Calls. This was done to make the Community Call 
notes more publicly available than we could make them previously.

## Helpful definitions

-   "LLVM Flang" is the Fortran 2018 compiler being developed as part of the 
LLVM project, and located in the LLVM monorepo ([github.com/llvm/llvm-project](http://github.com/llvm/llvm-project)).  

> -   "Classic Flang" is the Flang Fortran 2003/2008 compiler located at [github.com/flang-compiler/flang](http://github.com/flang-compiler/flang).

Reviewer: ```suggestion
-   "Classic Flang" is the Fortran 2003/2008 compiler located at [github.com/flang-compiler/flang](http://github.com/flang-compiler/flang).
```

Do we really want to include information about classic flang here? I am ok with leaving a reference to it somewhere else in the documentation, but it doesn't seem necessary here.

Author: I would prefer to leave the reference here primarily for disambiguation purposes. We still have users who convolve the two compilers and I like to be explicit about the terminology used in the calls and on the related documents.

Reviewer: Good point. Perhaps we could make it explicit that it is not **this** flang. No one really calls it LLVM Flang. Even we don't.

```suggestion
-   "Classic Flang" is a **different** Fortran 2003/2008 compiler located at [github.com/flang-compiler/flang](http://github.com/flang-compiler/flang).
```

## Call Information
- please see information in [GettingInvolved.md](https://github.com/llvm/llvm-project/blob/main/flang/docs/GettingInvolved.md) under the heading "Flang Biweekly Call"

## Flang Community

* **Websites:**  
  * **LLVM Flang:** [https://flang.llvm.org](https://flang.llvm.org)   
  * **Classic Flang:** [https://github.com/flang-compiler/flang](https://github.com/flang-compiler/flang) 

* **Issues:**  

>   * Issues for LLVM Flang should be reported in GitHub: [https://github.com/llvm/llvm-project/issues](https://github.com/llvm/llvm-project/issues).

Reviewer: ```suggestion
  * Issues for Flang should be reported in GitHub: [https://github.com/llvm/llvm-project/issues](https://github.com/llvm/llvm-project/issues).
```

  Use the "flang" tag for flang-related issues, please  
  * Issues for Classic Flang should be reported to the GitHub issues section at either [https://github.com/flang-compiler/flang/issues](https://github.com/flang-compiler/flang/issues)
  or [flang-compiler/flang-driver](https://github.com/flang-compiler/flang-driver) as appropriate

> * **LLVM mailing lists and Discourse channels:** There are now two communication channels hosted by llvm.org:  

Author: Is Discourse the preferred method, or is it Slack?  As far as I know, the community makes use of all of the communication channels listed in this document (Discourse, Slack, and the mailing list), and I am not totally sure that one has been given priority.  Personally, I prefer Discourse, but I know others have different preferences and I am loathe to put my personal view too far forward.

  * The Flang Discourse channel: [https://discourse.llvm.org/c/subprojects/flang/33](https://discourse.llvm.org/c/subprojects/flang/33)   

>   * The flang-commits mailing list: subscribe at [https://lists.llvm.org/cgi-bin/mailman/listinfo/flang-commits](https://lists.llvm.org/cgi-bin/mailman/listinfo/flang-commits) 

Reviewer: Sounds good.

* **Slack:** The [flang-compiler.slack.com](http://flang-compiler.slack.com)
workspace is used for informal real-time communication in the Flang community.  

>   * Classic Flang \- use the Slack channel \#classic-flang-pull-requests to

Reviewer: It is strange to have directions for pull requests for classic flang here.

 coordinate what pull requests you are working on

> * **LLVM Flang Development:**  

Reviewer: ### 
```suggestion
* **Flang Development:**  
```

>   * On GitHub: The source code for LLVM Flang is available on [https://github.com/llvm/llvm-project/tree/main/flang](https://github.com/llvm/llvm-project/tree/main/flang). It is recommended to start in the "docs" directory.  

Reviewer: ```suggestion
  * On GitHub: The source code for Flang is available on [https://github.com/llvm/llvm-project/tree/main/flang](https://github.com/llvm/llvm-project/tree/main/flang). It is recommended to start in the "docs" directory.  
```

  * Public plan: The team maintains a plan for upcoming Flang work at [https://github.com/orgs/flang-compiler/projects/](https://github.com/orgs/flang-compiler/projects/). This reflects current LLVM Flang priorities and context for the
  project and its long-term goals. (OpenMP work for flang, clang, and
  OpenMPIRBuilder is tracked separately.)

> * **Classic Flang Development:**  

Reviewer: I really don't think this belongs here.
